src/Indice.Features.Cases.AspNetCore/Controllers/AdminCasesController.cs

```diff
@@ -219,6 +219,26 @@ internal class AdminCasesController : ControllerBase
             return Ok(await _caseBookmarkService.GeUserActions(HttpContext.User, caseId));
         }
 
+        /// <summary>
+        /// Validate Case.
+        /// </summary>
+        /// <param name="caseId">The Id of the case.</param>
+        [Authorize(AuthenticationSchemes = CasesApiConstants.AuthenticationScheme, Policy = CasesApiConstants.Policies.BeSystemClient)]
+        [ProducesResponseType(200, Type = typeof(CaseValidationResponse))]
+        [Produces(MediaTypeNames.Application.Json)]
+        [HttpGet("validate-case/{caseId:guid}")]
+        public async Task<IActionResult> ValidateCase(Guid caseId) {
+            var @case = await _adminCaseService.GetCaseById(caseId);
+            if (@case is null) {
```

Comment:
θα έλεγα `if case is null or status  not Completed ==> NotFound`

και κατα την παραγωγή του QR Code θα ήταν καλό να το βαζουμε μονο στα completed